### PR TITLE
[WIP] Readiness checks for service using coslib

### PR DESCRIPTION
**Pull request by PietroPasotti**
Same as #21 but using https://github.com/canonical/cos-lib/pull/55

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -60,18 +58,10 @@ def __init__(self, *args):
             name="tempo",
             pebble_layer=self.generate_worker_layer,
             endpoints={"cluster": "tempo-cluster"},  # type: ignore
+            readiness_check_endpoint="http://localhost:3200/ready",
```
Comment (mmkay) [inline src/charm.py:61]: Might be worth checking if it will still work with TLS on. I wonder if http endpoint might start to refuse traffic then.

Comment (PietroPasotti) [reply src/charm.py:61]: uh, good thinking. will try that out

Comment (PietroPasotti) [reply src/charm.py:61]: image

ouch, you were right

Comment (PietroPasotti) [reply src/charm.py:61]: fix depends on https://github.com/canonical/cos-lib/pull/59

**Review by mmkay — APPROVED** [review]

**Review by michaeldmitry — APPROVED** [review]

**Pull request merged**